templates/components/engines/typescript/agent/tools.ts

```diff
@@ -1 +1,37 @@
-// Load tools here
\ No newline at end of file
+import { BaseTool } from "llamaindex";
+import config from "./tool_config.json";
+
+enum ExternalTool {
+  GoogleSearch = "google.GoogleSearchToolSpec",
+  Wikipedia = "wikipedia.WikipediaToolSpec",
+}
+
+type ToolConfig = { [key in ExternalTool]: Record<string, any> };
+
+export default class ToolFactory {
+  private static async createTool(
+    key: ExternalTool,
+    options: Record<string, any>,
+  ): Promise<BaseTool> {
+    if (key === ExternalTool.Wikipedia) {
+      const WikipediaTool = (await import("llamaindex")).WikipediaTool;
+      const tool = new WikipediaTool();
+      return tool;
+    }
+
+    // TODO: Implement GoogleSearchToolSpec
+
+    throw new Error(
+      `Sorry! Tool ${key} is not supported yet. Options: ${options}`,
+    );
+  }
+
+  public static async list(): Promise<BaseTool[]> {
```

Comment:
1. separate config from the factory 
2. How about moving the factory to `LlamaIndexTS` (might be useful for other use cases)? 

```suggestion
  public static async createTools(config: ToolConfig): Promise<BaseTool[]> {
```